eth/stagedsync/stage_execute.go

```diff
@@ -302,29 +299,31 @@ Loop:
 
 		updateProgress := batch.BatchSize() >= int(cfg.batchSize)
 		if updateProgress {
-			if err = batch.Commit(); err != nil {
-				return err
-			}
-			if !useExternalTx {
-				if err = s.Update(tx, stageProgress); err != nil {
-					return err
-				}
-				if err = tx.Commit(); err != nil {
+			if funcErr := func() error {
+				if err := batch.Commit(); err != nil {
 					return err
 				}
-				tx, err = cfg.db.BeginRw(context.Background())
-				if err != nil {
-					return err
+				if !useExternalTx {
+					if err := s.Update(tx, stageProgress); err != nil {
+						return err
+					}
+					if err := tx.Commit(); err != nil {
+						return err
+					}
+					tx, err := cfg.db.BeginRw(context.Background())
+					if err != nil {
+						return err
+					}
+					defer tx.Rollback()
```

Comment:
immediate rollback of just created transaction - means not possible to write anything in it anymore